### ISSUE TITLE
Feature Request: Support Setting file permissions on created log files

### ISSUE THREAD

**Issue by obviousrebel** [Delgan/loguru#401]
It would be nice to have a way to pass in permissions to use for created log file and if rotation, subsequent files would also have use these permissions. You can do this manually using os.chmod on the filename but it could be useful for loguru to have permissions passed in with a file sink to set permissions on current and rotated log files.

**Comment by obviousrebel** [comment]
Started a rough implementation on #403

**Comment by Delgan** [comment]
Hi @obviousrebel.

Thanks for your idea. I spent some time thinking about the implications of this suggestion.
Then I realized... It is already possible to customize file permissions!

While using [`logger.add()`](https://loguru.readthedocs.io/en/stable/api/logger.html#loguru._logger.Logger.add) for file sinks,  additional parameters are passed to the builtin [`open()`](https://docs.python.org/3/library/functions.html#open) function. What's great is that this function accepts an `opener` argument that can be used, among other things, to create the file with the desired permissions. Combined with the more low-level [`os.open()`](https://docs.python.org/3/library/os.html#os.open) function, it seems to work very well (also see [this example on StackOverflow](https://stackoverflow.com/a/45368120/2291710)).

```python
def opener(file, flags):
    return os.open(file, flags, 0o777)

logger.add("fooo/foo.log", rotation="100 kB", opener=opener)
```

What do you think of this workaround? Sure, it's a bit more verbose, but I really like this solution because it elegantly uses already existing arguments, it's quite simple and is generic enough to be applied to other cases (like `chown` and `chgrp`).

Also, thanks for having worked on a PR for this. We could keep the unit tests and add the above example to the documentation

**Comment by obviousrebel** [comment]
@Delgan Ah thats much cleaner! I will remove the added parameter and update docs to include this example. Thanks!

**Comment by Delgan** [comment]
Fixed by #403.
